src/api/resource.ts

```diff
@@ -64,83 +71,99 @@ export abstract class Resource {
     /**
      * Performs a query against this resource and subscribes to subsequent updates.
      */
-    protected reactiveRequest<T>(query: types.Query, path: string): Rx.Observable<T[]> {
+    protected reactiveRequest<T>(query: types.Query, path: string, options?: QueryOptions): Rx.Observable<T[]> {
         // We assume that the same query object on the same resource will always result in the same
         // underlying queryset (and therefore query observer).
         let serializedQuery = JSON.stringify([path, query]);
+        options = _.defaults({}, options || {}, {
+            reactive: false,
+        });
 
         return Rx.Observable.create<T[]>((observer) => {
-            let queryObserverId = this._queryObserverIdCache[serializedQuery];
-            let pendingQueries = this._pendingQueries[serializedQuery];
-
-            // Perform a REST query to get the observer identifier and to subscribe to new updates.
-            let subscriptions: Rx.Disposable[] = [];
-
-            if (queryObserverId) {
-                // This query observer identifier has already been cached. Check if it exists and in this
-                // case just subscribe to all items.
-                let queryObserver = this.connection.queryObserverManager().get(queryObserverId, false);
-                if (queryObserver) {
-                    if (queryObserver.status === QueryObserverStatus.INITIALIZED ||
-                        queryObserver.status === QueryObserverStatus.REINITIALIZING) {
-                        subscriptions.push(queryObserver.observable().subscribe(observer));
-                    }
+            if (options.reactive) {
```

Comment:
Should this rather be `if (!options.reactive) return;` to not go one level deeper?